docs/code-quality/ca2007.md

```diff
@@ -41,7 +41,9 @@ To fix violations, call <xref:System.Threading.Tasks.Task.ConfigureAwait%2A> on
 
 ## When to suppress warnings
 
-You can suppress this warning if you know that the consumer is not a graphical user interface (GUI) app or if the consumer does not have a <xref:System.Threading.SynchronizationContext>.
+This warning is intended for libraries, where the code may be executed in arbitrary environments and where code shouldn't make assumptions about the environment or how the caller of the method may be invoking or waiting on it. It is generally appropriate to suppress / disable the warning entirely over projects that represent application code rather than library code; in fact, running this analyzer on application code (for example, button click event handlers in a WinForms/WPF project) is likely to lead to the wrong actions being taken.
```

Comment:
```suggestion
This warning is intended for libraries, where the code may be executed in arbitrary environments and where code shouldn't make assumptions about the environment or how the caller of the method may be invoking or waiting on it. It is generally appropriate to suppress the warning entirely for projects that represent application code rather than library code; in fact, running this analyzer on application code (for example, button click event handlers in a WinForms or WPF project) is likely to lead to the wrong actions being taken.
```